lib/pejs.ts

```diff
@@ -173,27 +174,77 @@ export class PEJS {
+   * or whether you will use the selected Credentials, Presentation definition, evaluation results and/or presentation submission together with the signature options
+   *
+   * @param presentationDefinition the Presentation Definition
+   * @param selectedCredentials the PE-JS and/or User selected/filtered credentials that will become part of the Verifiable Presentation
    * @param signingCallBack the function which will be provided as a parameter. And this will be the method that will be able to perform actual
    *        signing. One example of signing is available in the project named. pe-selective-disclosure.
+   * @param options: Signing Params these are the signing params required to sign.
    *
-   * @return the signed presentations.
+   * @return the signed and thus Verifiable Presentation.
    */
-  public createVerifiablePresentation(
-    peJSPresentationDefinition: PresentationDefinition,
-    peJSSelectedCredentials: VerifiableCredential[],
-    peJSSigningKeyOptions: KeyPairOptions,
-    signingCallBack: (
-      presentationDefinition: PresentationDefinition,
-      selectedCredentials: VerifiableCredential[],
-      signingKeyOptions: KeyPairOptions
-    ) => Presentation[]
-  ): Presentation[] {
-    return signingCallBack(peJSPresentationDefinition, peJSSelectedCredentials, peJSSigningKeyOptions);
+  public verifiablePresentationFrom(
+    presentationDefinition: PresentationDefinition,
+    selectedCredentials: VerifiableCredential[],
+    signingCallBack: (callBackParams: PresentationSignCallBackParams) => VerifiablePresentation,
+    options: PresentationSignOptions
+  ): VerifiablePresentation {
+    const { holder, signatureOptions, proofOptions } = options;
+
+    function limitedDisclosureSuites() {
+      let limitDisclosureSignatureSuites: string[] = [];
+      if (proofOptions?.typeSupportsSelectiveDisclosure) {
+        if (!proofOptions?.type) {
+          throw Error('Please provide a proof type if you enable selective disclosure');
+        }
+        limitDisclosureSignatureSuites = [proofOptions.type];
+      }
+      return limitDisclosureSignatureSuites;
+    }
+
+    const holderDIDs: string[] = holder ? [holder] : [];
+    const limitDisclosureSignatureSuites = limitedDisclosureSuites();
+    this.evaluateCredentials(presentationDefinition, selectedCredentials, holderDIDs, limitDisclosureSignatureSuites);
+
+    const presentation = this.presentationFrom(presentationDefinition, selectedCredentials, holder);
+    const evaluationResults = this.evaluatePresentation(
+      presentationDefinition,
+      presentation,
+      limitDisclosureSignatureSuites
+    );
+    if (!evaluationResults.value) {
+      throw new Error('Could not get evaluation results from presentation');
+    }
+
+    const proof: Partial<Proof> = {
+      type: proofOptions?.type,
+      verificationMethod: signatureOptions?.verificationMethod,
+      created: proofOptions?.created ? proofOptions.created : new Date().toISOString(),
```

Comment:
I'm not sure that it's a good idea to have this as a parameter

Reply:
All low level library implementations do exactly the same. In the end I could always manipulate my clock on the holder side anyway. It is up to the verifier to check that the date is not in the future